### PR TITLE
Fix some issues caught by clang

### PR DESCRIPTION
**Pull request by adizaimi**
Fix some issues [and potential bugs] caught by warnings produced by clang in linux.

### CODE REVIEW

**Comment by adizaimi** [comment]
note, a more portable byteswap example can be found here: https://github.com/google/cityhash/blob/8af9b8c2b889d80c22d6bc26ba0df1afb79a30db/src/city.cc#L50

File: sqlite/src/vdbe.h [hunk 1]
```diff
@@ -35,7 +35,9 @@ typedef struct Vdbe Vdbe;
 ** The names of the following types declared in vdbeInt.h are required
 ** for the VdbeOp definition.
 */
+#if !defined(SQLITE_BUILDING_FOR_COMDB2)
 typedef struct sqlite3_value Mem;
```
Comment (mistachkin) [inline sqlite/src/vdbe.h]: I'm a bit confused about this change.  What was the warning this fixed?

Comment (adizaimi) [reply sqlite/src/vdbe.h]: double define of typedef (only C11 feature). Mem was defined in one of the includes.

**Comment by akshatsikarwar** [comment]
Please make sure this builds on macOS before merging.

**Pull request merged**